### DISCUSSION TITLE
Rekuperator ST-340

### DISCUSSION THREAD

**Discussion by djg-io** [mariusz-ostoja-swierczynski/tech-controllers#46] [Q&A]
Hi Mariusz,

I have the ST-340 recuperator controller. I can download data via php to Home Assistant but I can't control (change speed, turn on off, etc.). Any chance to add this model to your component?
I am sending json data:
`{"zones":{},"tiles":[{"id":1000,"parentId":0,"type":50,"menuId":0,"orderId":null,"visibility":true,"params":{"description":"Controller software version","version":"1.0.23","txtId":873,"iconId":95,"companyId":119,"controllerName":"ST-340","mainControllerId":0}},{"id":1425,"parentId":0,"type":1,"menuId":0,"orderId":null,"visibility":true,"params":{"description":"Temperature sensor","workingStatus":true,"txtId":4460,"value":12,"batteryLevel":null,"signalStrength":null}},{"id":1426,"parentId":0,"type":1,"menuId":0,"orderId":null,"visibility":true,"params":{"description":"Temperature sensor","workingStatus":true,"txtId":4461,"value":51,"batteryLevel":null,"signalStrength":null}},{"id":1427,"parentId":0,"type":1,"menuId":0,"orderId":null,"visibility":true,"params":{"description":"Temperature sensor","workingStatus":true,"txtId":3217,"value":169,"batteryLevel":null,"signalStrength":null}},{"id":1428,"parentId":0,"type":1,"menuId":0,"orderId":null,"visibility":true,"params":{"description":"Temperature sensor","workingStatus":true,"txtId":3216,"value":106,"batteryLevel":null,"signalStrength":null}},{"id":1440,"parentId":0,"type":6,"menuId":0,"orderId":null,"visibility":true,"params":{"description":"Universal status with widgets","statusId":1,"iconId":95,"widget1":{"txtId":5783,"value":0,"unit":-1,"type":1,"params":[0,0,0,0,0]},"widget2":{"txtId":0,"value":1,"unit":15,"type":1,"params":[0,0,0,0,0]}}},{"id":1441,"parentId":0,"type":6,"menuId":0,"orderId":null,"visibility":true,"params":{"description":"Universal status with widgets","statusId":1,"iconId":95,"widget1":{"txtId":5784,"value":0,"unit":-1,"type":1,"params":[0,0,0,0,0]},"widget2":{"txtId":0,"value":1,"unit":15,"type":1,"params":[0,0,0,0,0]}}},{"id":1448,"parentId":0,"type":41,"menuId":0,"orderId":null,"visibility":true,"params":{"description":"Date","year":2023,"month":1,"day":27,"dayId":5,"hours":17,"minutes":4}},{"id":2043,"parentId":0,"type":6,"menuId":2038,"orderId":null,"visibility":true,"params":{"description":"Universal status with widgets","statusId":1,"iconId":12,"widget1":{"txtId":5234,"value":0,"unit":-1,"type":1,"params":[0,0,0,0,0]},"widget2":{"txtId":0,"value":60,"unit":14,"type":1,"params":[0,0,0,0,0]}}},{"id":2070,"parentId":0,"type":40,"menuId":2070,"orderId":null,"visibility":true,"params":{"description":"Text information","statusId":422,"headerId":459,"iconId":102,"options":[]}},{"id":2092,"parentId":0,"type":40,"menuId":2092,"orderId":null,"visibility":true,"params":{"description":"Text information","statusId":5795,"headerId":594,"iconId":249,"options":[]}},{"id":2093,"parentId":0,"type":6,"menuId":2095,"orderId":null,"visibility":true,"params":{"description":"Universal status with widgets","statusId":0,"iconId":5,"widget1":{"txtId":0,"value":0,"unit":-1,"type":0,"params":[0,0,0,0,0]},"widget2":{"txtId":1843,"value":0,"unit":16,"type":1,"params":[0,0,0,0,0]}}},{"id":2094,"parentId":0,"type":6,"menuId":2094,"orderId":null,"visibility":true,"params":{"description":"Universal status with widgets","statusId":0,"iconId":95,"widget1":{"txtId":1845,"value":0,"unit":-1,"type":1,"params":[0,0,0,0,0]},"widget2":{"txtId":0,"value":0,"unit":-1,"type":1,"params":[0,0,0,0,0]}}}],"lastUpdate":"2023-01-27 17:05:07.218545+01"}`

**Comment by h4ckm4t** [comment]
Hi everyone,

I'm using an ST-340 V2 connected to a Wanas Recuperator and have been exploring how to integrate it with Home Assistant. Currently, I can only read sensor data with no control over the device. Here's what's accessible:

Bypass: State only
Additional Temp Sensor: State only
Party Mode: Detection only, no control
Room Temp: State only
Intake Temp: State only
Outflow Temp: State only
Outside Temp: State only
Seven txtld Sensors: Not functioning, showing "0"
Unfortunately, I'm unable to control any features like turning the device on/off or adjusting settings such as bypass and power. Control for the chiller (cooling)/heating would also be extremely useful.

Has anyone been able to successfully add control functions to this model or a similar setup? Any advice or updates on this topic would be greatly appreciated. It would be great to extend our capabilities beyond just monitoring these states.

Looking forward to any suggestions or progress in this area!
![Screenshot 2024-11-12 at 13 10 08](https://github.com/user-attachments/assets/c55b1085-2256-440b-962d-e3bb2d673822)

**Reply by vpontk** [reply]
> Hi everyone,
> 
> I'm using an ST-340 V2 connected to a Wanas Recuperator and have been exploring how to integrate it with Home Assistant. Currently, I can only read sensor data with no control over the device. Here's what's accessible:

Hi  h4ckm4t, would you be so kind  to share the modbus.yaml file? Or how do you connect to the control?

**Reply by h4ckm4t** [reply]
Hi,

First step is to integrate with the custom component from Mariusz: [tech-controllers] -> https://github.com/mariusz-ostoja-swierczynski/tech-controllers. 
That will give you the basic functionality, you’ll be able to pull sensor data from the ST-340 V2 into Home Assistant

But just so you know, there’s still no control over the recuperator itself. The integration i guess is mainly designed for floor heating controllers (Tech’s main business), so with the recuperator we’re stuck with read-only data......

That’s my current problem as well, I only see sensor status, which is annoying because I also have a chiller/aircon connected and would really like to switch it on/off depending on the recuperator state. Right now, there’s no clean way to do that without deeper Modbus register knowledge or extending the integration

**Reply by chuchlaster** [reply]
And what model of recuperator do you have?

**Reply by anarion80** [reply]
Modbus connection would be superior (because local). 

I am using my recuperator (Komfovent R-400-H) like so: https://github.com/anarion80/hass-config/blob/master/packages/komfovent_recuperator.yaml 

But you need to know the registers, like so: https://komfovent.com/en/downloads/MODBUS_connection_C6_EN.pdf

**Reply by chuchlaster** [reply]
I'm useing WANAS COMBO 630 XF and I have modbus registers and presets for it

See here: https://www.wanas.pl/wp-content/uploads/2025/04/Instrukcja-instalacji-i-obslugi-Combo-630.pdf

**Reply by h4ckm4t** [reply]
Hi @chuchlaster 

I have a WANAS 555 XF connected to a WANAS Maxi R32 chiller. My setup is very similar to yours, and I’d really like to take control over the recuperator instead of just reading sensor data

Could you support me with a step-by-step guide to set up the Modbus connection? I’d really appreciate it – even basic instructions on registers, wiring, and HA configuration would help a lot

Thanks in advance!

**Reply by chuchlaster** [reply]
You need some RS232/485 to wifi converter (I'm useing Waveshare) and some 4-channel potential-free relay - connections and modbus parametres and registers you have in INSTALLATION AND OPERATING MANUAL of your recuperator.

And one thing - If you have wifi connection with you recuperator you have to unplug it. You can use either modbus or wifi - not both...

Potential free realy is used to turn on 1 and 3 airflow rate and for "open window" mode

All ports on recuperator main board should be described

**Reply by h4ckm4t** [reply]
> You need some RS232/485 to wifi converter (I'm useing Waveshare) and some 4-channel potential-free relay - connections and modbus parametres and registers you have in INSTALLATION AND OPERATING MANUAL of your recuperator.
> 
> And one thing - If you have wifi connection with you recuperator you have to unplug it. You can use either modbus or wifi - not both...
> 
> Potential free realy is used to turn on 1 and 3 airflow rate and for "open window" mode
> 
> All ports on recuperator main board should be described

Hi @chuchlaster

Thanks a lot for the inspiration 🙏 the Modbus works super well now...

Quick question: what potential-free relay are you using? Do you have a specific model? I’ll probably need to go Tasmota or Zigbee since I don’t have any extra cables running up to the attic where the recuperator is… and via Modbus there is no rational way to control the fans :( 
Cheers!
